app/services/github/client.rb

```diff
@@ -0,0 +1,98 @@
+module Github
+  # Github client (uses ocktokit.rb as a backend)
+  class Client
+    class << self
+      # adapted from https://api.rubyonrails.org/classes/Module.html#method-i-delegate_missing_to
+      def method_missing(method, *args, &block)
+        return super unless target.respond_to?(method, false)
+
+        # define for re-use
+        self.class.define_method(method) do |*new_args, &new_block|
+          request do
+            target.public_send(method, *new_args, &new_block)
+          end
+        end
+
+        # call the original method, this will only be called the first time
+        # as in subsequent calls, the newly defined method will prevail
+        request do
+          target.public_send(method, *args, &block)
+        end
+      end
+
+      # adapted from https://api.rubyonrails.org/classes/Module.html#method-i-delegate_missing_to
+      def respond_to_missing?(method, _include_all = false)
+        target.respond_to?(method, false) || super
+      end
+
+      private
+
+      def request
+        Honeycomb.add_field("app.name", "github.client")
+        yield
+      rescue Octokit::Error => e
+        class_name = e.class.name.demodulize
```

Comment:
This seems long enough to be replaced by `handle_errror(e)` or something.

Reply:
makes sense